pennylane/tape/tapes/tape.py

```diff
@@ -339,6 +361,23 @@ def _update_circuit_info(self):
         )
         self.num_wires = len(self.wires)
 
+    def _update_observables(self):
+        """Update information about observables, including the wires that are acted upon and
+        identifying any repeated observables"""
+        obs_wires = [wire for m in self.measurements for wire in m.wires if m.obs is not None]
+        self._repeated_observables = []
+        self._repeated_observables_id = []
+
+        if len(obs_wires) != len(set(obs_wires)):
```

Comment:
just to learn: how come the lengths of ``obs_wires`` and ``set(obs_wires)`` can be different?

Reply:
`obs_wires` gives all of the wires from all of the observables. If we have two observables on the same wire, this might be e.g. `[0, 1, 2, 0]`. The `set(obs_wires)` removes the repetitions, so would have length 3.